kubernetes/k8s-listener.py

```diff
@@ -388,6 +439,88 @@ def main():
         if c != '\r' or c2 != '\n':
             raise "Expected CRLF, got %c%c" % (c, c2)
 
+def get_romana_hosts():
+    """
+    returns a list of romana host IPs
+    
+    Example: ["10.0.0.1", 10.1.0.1" ]
+    """
+
+    romana_hosts = []
+    r = requests.get(topology_url)
+    print 'Topology service returned %s' % r.content
+    hosts = simplejson.loads(r.content)
+
+    # romana_ip is a CIDR like 10.0.0.1/16
+    # we only want ip part
+    for host in hosts:
+        mask_idx = host["romana_ip"].index("/")
+        romana_hosts.append(host["romana_ip"][:mask_idx])
+    return romana_hosts
+
+# We want to receive json object as a POST.
+from BaseHTTPServer import BaseHTTPRequestHandler,HTTPServer
+from mimetools import Message
+from StringIO import StringIO
+PORT_NUMBER = 9630
+HTTP_Unprocessable_Entity = 422
+class AgentHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type','text/html')
+        self.end_headers()
+        # Send the html message
+        self.wfile.write("Hello Get !")
+        return
+
+    def do_POST(self):
+        """
+        Processes POST requests
+        extracts romana policy definition objects and passes it down for implementation
```

Comment:
Docstrings are usually formatted a little differently. Since this is temporary code, we can let it slide, though.